src/admin/BulkUserCreationPage.test.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import http from 'axios';
+import { render, waitFor, fireEvent } from '@testing-library/react';
+
+import { BulkUserCreationPage } from '.';
+import { useAuthentication } from '../authentication/context';
+import { User } from '../api';
+
+jest.mock('axios');
+jest.mock('../authentication/context');
+
+describe('Bulk user creation page', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('creates users with emails from comma-separated input and selected role and shows a success or error message', async () => {
```

Comment:
It's a big test, but with React Testing Library's integration-rather-than-unit style, I've found longer tests to be acceptable, as we reduce repetition, over-abstraction, and still manage to find out what went wrong easily thanks to its great logging.